color.py

```diff
@@ -18,6 +18,38 @@ def intra_hom_fraction(g3d_particle, nearby_g3d_particles, max_separation):
         return None
     return float(num_intra_hom) / num_g3d_particles
 
+def same_haplotype_fraction(g3d_particle, nearby_g3d_particles):
+    hom_name = g3d_particle.get_hom_name()
+    haplotype = g3d_particle.get_haplotype()
+    num_g3d_particles = -1 # to exclude self
+    num_same_haplotype = -1 # to exclude self
+    for nearby_g3d_particle in nearby_g3d_particles:
+        num_g3d_particles += 1
+        nearby_hom_name = nearby_g3d_particle.get_hom_name()
+        nearby_haplotype = nearby_g3d_particle.get_haplotype()
+        if nearby_haplotype == haplotype:
+                num_same_haplotype += 1
+    if num_g3d_particles == 0:
+        return None
+    return float(num_same_haplotype) / num_g3d_particles
+
+
+def same_haplotype_fraction_excluding_same_homolog(g3d_particle, nearby_g3d_particles):
+    hom_name = g3d_particle.get_hom_name()
+    haplotype = g3d_particle.get_haplotype()
+    num_g3d_particles = -1 # to exclude self
+    num_same_haplotype = -1 # to exclude self
```

Comment:
Please initialize num_same_haplotype to 0 (rather than -1), because the self particle won't be counted in this (excluding same homolog) case.

Reply:
And also delete the "# to exclude self" comment.